### ISSUE TITLE
Tmux php fatal error for nzedb.log permissions issue

### ISSUE THREAD

**Issue by Komorebi-E** [nZEDb/nZEDb#1598]
Server setup based on pull on 21/02/2015, Ubuntu 14.04 LTS.

Below is the error from the tmux views for 1:utils* and 2:post#-

The fix to get rid of the symptom is below, however I would assume that when the logging process creates the log it assigns incorrect permissions.

    sudo chmod 0664 /var/www/nZEDb/resources/logs/nzedb.log

    "PHP Fatal error: Uncaught exception 'LoggerException' with message 'Unable to open log file /var/www/nZEDb/resources/logs/nzedb.log' in /var/www/nZEDb/nzedb/controllers/Logger.php:456
    Stack trace
    #0 /var/www/nZEDb/nzedb/controllers/Logger.php(410) : Logger->openFile()
    #1 /var/www/nZEDb/nzedb/controllers/Logger.php(200) : Logger->setlogFile()
    #2 /var/www/nZEDb/nzedb/controllers/processing/PostProcess.php(95) : Logger->__construct(Array)
    #3 /var/www/nZEDb/misc/update/nix/tmux/bin/groupfixrelnames.php(139) : nzedb\processing\PostProcess_>__construct(Array)
    #4 {main}
    thrown in /var/www/nZEDb/nzedb/controllers/Logger.php on line 456

Group membership

    getent group | grep www
    www-data:x:33:komorebi,nzedb

    komorebi@odroid-server:/var/www/nZEDb/resources/logs$ ls -l
    total 3444
    -rw-rw-r-- 1 komorebi     komorebi     2379537 Feb 21 13:00 decryptHashes-20150221.log
    -rw-rw-r-- 1 komorebi     komorebi      193052 Feb 21 13:00 fixReleaseNames-20150221.log
    -rw-r--r--  1 www-data    www-data      56 Feb 21 11:12 nzedb.log
    -rw-rw-r-- 1 komorebi     komorebi      525440 Feb 21 13:00 postprocessing_additional-20150221.log
    -rw-rw-r-- 1 komorebi     komorebi       16264 Feb 21 11:45 sequential-20150221.log
    -rw-rw-r-- 1 komorebi     komorebi      385986 Feb 21 13:00 update_releases-20150221.log

I fixed the symptom on my system by changing the permissions:

    sudo chmod 0664 nzedb.log

**Comment by kevinlekiller** [comment]
The web and CLI users read and write from this file.

You need to properly set up your web and CLI users to be able to read and write to this file.

Usually this is done by creating or using a common group and adding your CLI and web users to this group, then you give permission to that group to read and write this file.

For example you can add user komorebi to group www-data, then you chown the group of the file to www-data and give r/w to the file for group level.

**Comment by Komorebi-E** [comment]
Hello,

Thank you for the response. Your solution is what I did, komorebi and nzedb, had already been added to www-data when I was building the server. However the log file that is created by the logging process was created with no group write permission, I had to manually set it to stop the error.

It may be a difference in expectations. As I don't have to touch or create the file to create it for the logging process and it isn't mentioned in the installation guide, I expected it to be set up with group write. I may be wrong in having this expectation.
